src/main/scala/akka/persistence/jdbc/journal/JdbcAsyncWriteJournal.scala

```diff
@@ -34,9 +35,18 @@ import slick.jdbc.JdbcBackend._
 import scala.collection.immutable._
 import scala.concurrent.{ ExecutionContext, Future }
 import scala.util.{ Failure, Success, Try }
+import akka.pattern.pipe
 
 object JdbcAsyncWriteJournal {
   private case class WriteFinished(pid: String, f: Future[_])
+  /**
+   * Extra Plugin API: May be used to issue in-place updates for events.
+   * To be used only for data migrations such as "encrypt all events" and similar operations.
+   *
+   * The write payload may be wrapped in a [[akka.persistence.journal.Tagged]],
+   * in which case the new tags will overwrite the existing tags of the event.
+   */
+  final case class InPlaceUpdateEvent(pid: String, seqNr: Long, write: AnyRef)
```

Comment:
The parameter names can be improved. I would prefer renaming `pid` to `persistenceId`, and `seqNr` to `sequenceNr` (to keep the names consistent with the rest of the apis). Also I would suggest renaming `write` to something like `updatedPayload` or `replacementPayload` (or replacementMessage/replacementEvent, not sure what is best).

Reply:
Good suggestions, thanks!

Reply:
Also, I was thinking, should the type of `write` be `Any` or `AnyRef`? In most other places the type of the message seems to be `Any`. Perhaps this also solved the boxing issues you mentioned in Patriks comment

Reply:
Yes, the type of the payload in`PersistentRepr` is `Any`. Although I agree that an event is supposed to be an `AnyRef`, the current API allows it to be `Any` (and as consequence also `AnyVal`).